spec/Appendix B -- Grammar Summary.md

```diff
@@ -331,6 +343,8 @@ TypeSystemDirectiveLocation : one of
   `ARGUMENT_DEFINITION`
   `INTERFACE`
   `UNION`
+  `TAGGED`
+  `TAGGED_MEMBER_FIELD_DEFINITION`
```

Comment:
Should we distinguish this from `FIELD_DEFINITION` and `ARGUMENT_DEFINITION`? I can't immediately think of a use case for a directive that would apply on normal fields or normal arguments, but would be incoherent on a tagged member field?

If we *don't* define it specially, that weighs in on the input-vs-output question as well; for a tagged member field, which directives does it permit? Field directives or argument directives? If it's to be used in both input and output, it has to permit both; is that reasonable?

Reply:
`FIELD_DEFINITION`, `ARGUMENT_DEFINITION` and `INPUT_FIELD_DEFINITION` have separate entries because they are separate things, tagged member fields are also separate to all of these, they're neither FIELD nor INPUT_FIELD, so I think it makes sense to have a separate directive location for them. As you rightly point out member fields are different because they could be for input _or_ output, so it's clear that using directives targetted at fields or arguments may not make sense for tagged, so again differentiation is critical.

Reply:
> I can't immediately think of a use case for a directive that would apply on normal fields or normal arguments, but would be incoherent on a tagged member field?

```graphql
directive @required(with: String!) on ARGUMENT_DEFINITION, INPUT_FIELD_DEFINITION

input UpdateUser {
  name: String
  password: String
  passwordConfirmation: String @required(with: "password")
}
```